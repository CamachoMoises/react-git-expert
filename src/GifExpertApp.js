import React, { useState } from 'react';
import {AddCategory} from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

const GifExpertApp = ({defaultCategories=[]}) => {
	// const categories = [' One Punch', 'Samurai X', 'Dragon Ball'];
	const [categories, setCategories] = useState(defaultCategories);
	// const handleAdd = () => {
	// 	// setCategories([...categories, 'Naruto']);
    //     setCategories(cats => ['One Piece',...cats])
	// };
	return (
		<>
			<h2> GifExpertApp </h2>
			<AddCategory setCategories={setCategories} />
            <hr />
			<ol>
				{categories.map((category) =>
                <GifGrid 
                    category={category} 
                    key={category}
                />
                )}
			</ol>
		</>
	);
};

export default GifExpertApp;
