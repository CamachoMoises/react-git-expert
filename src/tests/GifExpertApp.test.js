import { shallow } from 'enzyme';
import React from 'react';
import GifExpertApp from '../GifExpertApp.js';

describe('Test in GifExpertApp component', () => {
	test('should ', () => {
		const wrapper = shallow(<GifExpertApp />);
		expect(wrapper).toMatchSnapshot();
	});
	test('should show a list of categories', () => {
		const categories = ['Batman', 'Superhero'];
		const wrapper = shallow(<GifExpertApp defaultCategories={categories} />);
		expect(wrapper.find('GifGrid').length).toBe(categories.length);
	});
});
