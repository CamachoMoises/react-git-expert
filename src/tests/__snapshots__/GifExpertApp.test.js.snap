// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in GifExpertApp component should  1`] = `
<Fragment>
  <h2>
     GifExpertApp 
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
